File: src/main/java/de/simpletactics/wiki/lib/service/port/WikiPort.java
package de.simpletactics.wiki.lib.service.port;

import de.simpletactics.wiki.lib.model.WikiType;

public interface WikiPort {

  String getTopicsAsJson(int id, String username);

  <T> T getTopicsAsObject(int id, String username);

  void updateEntry(String topic, String content, int id);

  void updateTopic(String topic, int id);

  int addEntry(String topic, String content, String threadId, WikiType wikiType);

  int addTopic(String id, String topic);

  void deleteTopic(int id);

  void deleteEntry(int id);

  WikiType getWikiType(int id);

}

File: src/main/kotlin/de/simpletactics/wiki/lib/services/port/WikiPort.kt
package de.simpletactics.wiki.lib.services.port

import de.simpletactics.wiki.lib.adapter.dto.EntryEntity
import de.simpletactics.wiki.lib.adapter.dto.TopicEntity
import de.simpletactics.wiki.lib.model.WikiType

interface WikiPort {

    fun getTopic(id: Int, hasAccess: () -> Boolean): TopicEntity?

    fun createTopic(parentId: Int, topic: String, hasAccess: () -> Boolean): Int

    fun updateTopic(id: Int, topic: String, hasAccess: () -> Boolean): Int

    fun deleteTopic(id: Int, hasAccess: () -> Boolean)

    fun getEntry(id: Int, hasAccess: () -> Boolean): EntryEntity?

    fun createEntry(topicId: Int, headline: String, body: String, hasAccess: () -> Boolean): Int

    fun updateEntry(id: Int, headline: String, body: String, hasAccess: () -> Boolean): Int

    fun deleteEntry(id: Int, hasAccess: () -> Boolean)

    fun getWikiType(id: Int, hasAccess: () -> Boolean): WikiType?

}
